### PR TITLE
Enable grpc gzip compression

### PR DESCRIPTION
**Pull request by bryanlb**
Enables server-side compression per https://github.com/grpc/grpc-java/blob/d4fa0ecc07495097453b0a2848765f076b9e714c/examples/src/main/java/io/grpc/examples/experimental/CompressingHelloWorldServerAllMethods.java#L48

### CODE REVIEW

File: kaldb/src/test/java/com/slack/kaldb/testlib/MessageUtil.java [hunk 1]
```diff
@@ -62,7 +63,12 @@ public static LogWireMessage makeWireMessage(int i, String ts) {
     String id = DEFAULT_MESSAGE_PREFIX + i;
     Map<String, Object> fieldMap = new HashMap<>();
     fieldMap.put(LogMessage.ReservedField.TIMESTAMP.fieldName, ts);
-    String message = String.format("The identifier in this message is %s", id);
+    String message =
+        String.format(
+            "The identifier in this message is %s - %s",
+            id,
+            // make example strings long for easier message compression visibility
+            RandomStringUtils.randomAlphabetic(8000));
```
Comment (mansu) [inline kaldb/src/test/java/com/slack/kaldb/testlib/MessageUtil.java:71]: Let's drop this string.

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java [hunk 2]
```diff
@@ -122,7 +122,8 @@ private void updateStubs() {
   }
 
   private KaldbServiceGrpc.KaldbServiceFutureStub getKaldbServiceGrpcClient(String server) {
-    return Clients.newClient(server, KaldbServiceGrpc.KaldbServiceFutureStub.class)
+    return GrpcClients.builder(server)
+        .build(KaldbServiceGrpc.KaldbServiceFutureStub.class)
```
Comment (mansu) [inline kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:126]: Add a comment stating that this is a client side compression and we need to set the server side separately.

File: kaldb/src/main/java/com/slack/kaldb/server/ArmeriaService.java [hunk 3]
```diff
@@ -113,7 +117,20 @@ public Builder withGrpcService(BindableService grpcService) {
...
-              .useBlockingTaskExecutor(true);
+              .useBlockingTaskExecutor(true)
+              .intercept(
+                  new ServerInterceptor() {
+                    @Override
+                    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+                        ServerCall<ReqT, RespT> call,
+                        Metadata headers,
+                        ServerCallHandler<ReqT, RespT> next) {
+                      // sets gzip message encoding per
```
Comment (mansu) [inline kaldb/src/main/java/com/slack/kaldb/server/ArmeriaService.java:128]: Please add a comment this is for response side encryption.

**Review by mansu — CHANGES_REQUESTED** [review]

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java (hunk 2 above)
Comment (bryanlb) [reply kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:126]: Done.

File: kaldb/src/main/java/com/slack/kaldb/server/ArmeriaService.java (hunk 3 above)
Comment (bryanlb) [reply kaldb/src/main/java/com/slack/kaldb/server/ArmeriaService.java]: Done.

**Review by mansu — APPROVED** [review]

**Pull request merged**